src/praisonai-agents/praisonaiagents/tools/gather_tools.py

```
@@ -0,0 +1,272 @@
+            json={"purpose": "post"},
+            timeout=15,
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        challenge, difficulty = data["challenge"], data["difficulty"]
+
+        for nonce in range(MAX_POW_ITERATIONS):
+            hash_bytes = hashlib.sha256(f"{challenge}:{nonce}".encode()).digest()
+            if int.from_bytes(hash_bytes[:4], "big") >> (32 - difficulty) == 0:
+                return {"pow_challenge": challenge, "pow_nonce": str(nonce)}
+        return None
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)